client/components/Checkbox.js

```diff
@@ -0,0 +1,51 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+export default class Checkbox extends React.PureComponent {
+  render () {
+    const {
+      className,
+      checked,
+      readOnly,
+      label,
+      name,
+      value,
+      onCheck
+    } = this.props;
+
+    let checkbox = (
+      <span className={className}>
+        <input
+          type="checkbox"
+          id={name}
```

Comment:
Since this is a reusable component, its id should be as unique as possible, so as to not conflict with any other checkboxes that might be on the page. One way to do this would be to have some counter in the top level of the module, and add a property to the component that increments that, then create an id from that. Something like the following:

```jsx
let componentId = 0;

export default class Checkbox extends React.PureComponent {
  constructor() {
    this.componentId = componentId++;
  }

  render() {
    const id = `checkbox-${this.props.name}-${this.componentId}`;
    return (
      <input id={id}/>
    );
  }
}
```

This is particularly true because the component may be created without a name (the name prop is optional).

Reply:
Done, passed id through map(i) parameter